File: backend/dtos/src/main/java/io/mateu/dtos/Element.java
package io.mateu.dtos;

import java.util.Collections;
import java.util.Map;

/** Metadata for a html element */
public record Element(String name, Map<String, Object> attributes, String content)
    implements ComponentMetadata {

  public Element {
    attributes = attributes != null ? Collections.unmodifiableMap(attributes) : Map.of();
  }

  @Override
  public Map<String, Object> attributes() {
    return Collections.unmodifiableMap(attributes);
  }
}

File: frontend/web/monorepo/libs/mateu/src/mateu/shared/apiClients/dtos/componentmetadata/Element.ts
import ComponentMetadata from "@mateu/shared/apiClients/dtos/ComponentMetadata";

export default interface Element extends ComponentMetadata {
    name: string
    content: string
    attributes: Record<string, any>
    on: Record<string, string>
    // When true, `content` is injected as raw HTML (innerHTML) instead of escaped text.
    html?: boolean
}
